### PR TITLE
Adjust the params of the access token request when basic auth is enabled

### PR DESCRIPTION
**Pull request by GeorgianaElena**
Fixes https://github.com/jupyterhub/oauthenticator/issues/565

### CODE REVIEW

File: oauthenticator/oauth2.py [hunk 1]
```diff
@@ -561,11 +561,18 @@ def build_access_tokens_request_params(self, handler, data=None):
...
             "data": data,
         }
+
+        # the client_id and client_secret should not be included in the access token request params
+        # when basic authentication is used
+        # ref: https://www.rfc-editor.org/rfc/rfc6749#section-2.3.1
+        if self.basic_auth:
+            params.update(
+                [("client_id", self.client_id), ("client_secret", self.client_secret)]
+            )
```
Comment (consideRatio) [inline oauthenticator/oauth2.py:574]: Should this be `if not ...`?

Comment (GeorgianaElena) [reply oauthenticator/oauth2.py:574]: haha, of course :D

**Comment by consideRatio** [comment]
@GeorgianaElena is this a bugfix of a regression introduced in the refactoring PR, or something more as well?

**Comment by consideRatio** [comment]
Ah, I think I understand.

Before refactoring PR we had a bug
After refactoring PR we had another bug
Now this is fixed fully

**Comment by GeorgianaElena** [comment]
I think it's more. I believe, according to https://github.com/jupyterhub/oauthenticator/issues/273, the GenericOAuthenticator (and possibly other oauthenticators also) wasn't including the client creds in the request at all.

So, the refactor, added the creds to the request for _all_ authenticators, but didn't take into account the basic_auth thing.

So, now this fixes that. 

I believe labeling it as a bug, is enough? 

Anyway, after we get https://github.com/jupyterhub/oauthenticator/pull/561 merged, I will focus on documenting the refactorization as best as I can, so we can make a release and see if it actually broke more things than expected 😬

**Pull request merged**